### PR TITLE
STONEBLD-914: reduce scope to repositories in the cluster

### PR DESCRIPTION
**Pull request by Michkov**
- Do not go via all repositories available to GitHub app but process only repositories which match to component
- Separate configuration per installation
- [STONEBLD-915](https://issues.redhat.com//browse/STONEBLD-915): respect baseBranch based on Component revision

### CODE REVIEW

File: controllers/git_tekton_resources_renovater_test.go [hunk 1]
```diff
@@ -52,28 +53,82 @@ var _ = Describe("Git tekton resources renovater", func() {
...
+
+			// This is one installation - two matching repos
+			componentNamespacedName1 := createComponentForPaCBuild(getComponentData(componentConfig{componentKey: types.NamespacedName{Name: "test1"}, gitURL: "https://github/test1/repo1"}))
+			componentNamespacedName2 := createComponentForPaCBuild(getComponentData(componentConfig{componentKey: types.NamespacedName{Name: "test2"}, gitURL: "https://github/test1/repo2"}))
+			// Second installation
+			componentNamespacedName3 := createComponentForPaCBuild(getComponentData(componentConfig{componentKey: types.NamespacedName{Name: "test3"}, gitURL: "https://github/test2/repo1"}))
+			// Third installation
+			componentNamespacedName4 := createComponentForPaCBuild(getComponentData(componentConfig{componentKey: types.NamespacedName{Name: "test4"}, gitURL: "https://github/test3/repo2"}))
+			// Set 2 installations per job
+			os.Setenv(InstallationsPerJobEnvName, "2")
```
[the remark this answers is not in the document]
Comment (Michkov) [reply controllers/git_tekton_resources_renovater_test.go:125]: updated, added to afterall

File: controllers/git_tekton_resources_renovater.go [hunk 2]
```diff
@@ -114,35 +131,67 @@ func (r *GitTektonResourcesRenovater) Reconcile(ctx context.Context, req ctrl.Re
...
+				repositories: repositories,
+			})
 	}
 
-	for i := 0; i < len(installations); i += InstallationsPerJob {
-		end := i + InstallationsPerJob
+	// Generate renovate jobs. Limit processed installations per job.
+	var installationPerJobInt int
+	installationPerJobStr := os.Getenv(InstallationsPerJobEnvName)
+	if regexp.MustCompile(`\d`).MatchString(installationPerJobStr) {
```
[the remark this answers is not in the document]
Comment (Michkov) [reply controllers/git_tekton_resources_renovater.go]: updated to `^\d+$`

[the remark this answers is not in the document]
Comment (tkdchen) [reply controllers/git_tekton_resources_renovater.go]: The following for-loop will be infinite loop when 0 is set.

[the remark this answers is not in the document]
Comment (Michkov) [reply controllers/git_tekton_resources_renovater.go]: added check on 0

**Review by tkdchen — APPROVED** [review]

File: pkg/github/github_client.go [hunk 3]
```diff
@@ -153,9 +154,15 @@ func getInstallations(appId int64, privateKeyPem []byte) ([]ApplicationInstallat
 				// TODO analyze the error
 				continue
 			}
+			installationClient := NewGithubClient(token.GetToken())
+			repositoriesList, _, err := installationClient.client.Apps.ListRepos(context.Background(), &github.ListOptions{})
```
Comment (mmorhun) [inline pkg/github/github_client.go:158]: Do we need to respect pagination  here?

Comment (Michkov) [reply pkg/github/github_client.go]: added

File: controllers/git_tekton_resources_renovater.go [hunk 4]
```diff
@@ -114,35 +131,70 @@ func (r *GitTektonResourcesRenovater) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{}, nil
 	}
 	privateKey := pacSecret.Data[gitops.PipelinesAsCode_githubPrivateKey]
-	installations, slug, err := github.GetInstallations(githubAppId, privateKey)
+	githubInstallations, slug, err := github.GetInstallations(githubAppId, privateKey)
```
[the remark this answers is not in the document]
Comment (Michkov) [reply controllers/git_tekton_resources_renovater.go]: updated

**Comment by Michkov** [comment]
`The step "get-vulnerabilities" in TaskRun "test-component-pac-rpvx-on-pull-request-qmtrd-clair-scan" failed to pull the image "". The pod errored with the message: "Back-off pulling image "quay.io/redhat-appstudio/clair-in-ci:latest"."'`

/test build-service-e2e

File: controllers/git_tekton_resources_renovater.go [hunk 5]
```diff
@@ -114,35 +131,70 @@ func (r *GitTektonResourcesRenovater) Reconcile(ctx context.Context, req ctrl.Re
...
+		return ctrl.Result{}, err
 	}
-	if err := r.Client.Delete(ctx, &configmap); err != nil {
-		if !errors.IsNotFound(err) {
-			return ctrl.Result{}, err
+	componentUrlToBranchMap := make(map[string]string)
+	for _, component := range componentList.Items {
+		if component.Spec.Source.GitSource != nil {
+			url := strings.TrimSuffix(strings.TrimSuffix(component.Spec.Source.GitSource.URL, ".git"), "/")
+			componentUrlToBranchMap[url] = component.Spec.Source.GitSource.Revision
```
[the remark this answers is not in the document]
Comment (Michkov) [reply controllers/git_tekton_resources_renovater.go:149]: yes then it's empty and it's processed properly later in code.

**Pull request merged**